### ISSUE TITLE
Add support to set proxy details from the script

### ISSUE THREAD

**Issue by acandylevey** [a85/POSTMan-Chrome-Extension#5394]
Please read through the [guidelines](https://github.com/postmanlabs/postman-app-support#guidelines-for-reporting-issues) before creating a new issue.

**Is your feature request related to a problem? Please describe.**
It is frustrating when dealing with proxies if where we have a couple of different ones we use, and not all requests are required to use them. 

**Describe the solution you'd like**
The ability to set the proxy settings from environment variables or at least least change them on the fly from test scripts / pre-request scripts along with ability to turn them on/off

**Describe alternatives you've considered**
Manually turning them on / off for the request - Which makes it incompatible with the test runner.

**Additional context**
Being able to it something like below would be perfect, as it could be done on a per collection or a per request basis as needed and could pull from either global or environment variables to set
![image](https://user-images.githubusercontent.com/6847665/47278779-4e3ccf00-d617-11e8-93c8-76ae968047b9.png)

**Comment by rsarunprashad** [comment]
I am looking for this  a while, dynamic proxy ennoblement through scripts

**Comment by bvic-dev** [comment]
I'm also waiting for this, it will help a lot :)

**Comment by jalal-jafari** [comment]
It's a nice feature to have, waiting to be added.

**Comment by mnunezdm** [comment]
hope this gets implemented soon

**Comment by CainDS** [comment]
Any time frame on when this will be done? We've ran into a bunch of problems with test worldpay APIs needing a certain proxy to authenticate and another calls need a separate one. Currently no way around this when running through newman so was hoping to be able to do it at a script level.

**Comment by SpenceUK** [comment]
Hi, this feature would be very useful. Example I have is working remotely I have a collection with multiple requests but I have to go into the settings and manually turn the proxy on or off as required for the individual requests. I should be able to automate this task in a pre-request script.

Is this feature in active development?

**Comment by wenditer** [comment]
I'm also looking for this feature. Some APIs that I call require proxy, some APIs don't require, I have to constantly enable/disable proxy in settings, which is frustrating. Also I cannot automate such requests, I cannot connect output of one request to the input of another request and make them runnable together as a single one, because one request requires proxy and another requires absence of proxy.

**Comment by jcarenas** [comment]
Same expectation here

**Comment by mottersheadt** [comment]
Agreed. This will be a very useful features when working with proxy servers in different environments.

**Comment by simondiesenreiter** [comment]
Such a feature would be very helpful +1

**Comment by steel119** [comment]
+1

**Comment by noise-trader** [comment]
+1

**Comment by yapxue** [comment]
+1

**Comment by d4ndz** [comment]
+1

**Comment by RickyJam** [comment]
+1

**Comment by deneboulton** [comment]
+1

**Comment by taotao365s** [comment]
+1
